src/v1/index.js

```diff
@@ -422,10 +422,6 @@ function appendChildOrInsertBefore (host, newNode, refNode) {
   switch (nodeType) {
     case 'fragment':
     case 'node':
-      if (canPatchNativeAccessors) {
```

Comment:
Just removing this fixed the issue. All other tests still pass. What was happening is that the `children` of the `DocumentFragment` weren't being updated after being moved. The `addNodeToNode` automatically does this using `registerNode`.